src/librustc/diagnostics.rs

```diff
@@ -168,6 +168,11 @@ match x {
 ```
 "##,
 
+E0010: r##"
+The value of static and const variables must be known at compile time, and they
+live for the entire lifetime of a program. They can not use allocation.
```

Comment:
I think a longer comment about allocation on the heap being a run-time operation would be ideal. Mentioning `Box` values could also be good but the core concept is allocation.